src/layout/List/ListComponent.tsx

```diff
@@ -127,67 +131,85 @@ export const ListComponent = ({
     return JSON.stringify(rowAsValue(datalist));
   };
 
-  const createLabelRadioButton = (datalist) => {
+  const createLabelRadioButton = (datalist, headers) => {
     let label = '';
-    for (const key in formData) {
-      label += `${key} ${datalist[key]} `;
+    let index = 0;
+    for (const key in datalist) {
+      label += `${headers[index]} ${datalist[key]} `;
+      index++;
     }
     return label;
   };
 
+  const classes = useRadioStyles();
+  const RenderLegend = legend;
+
   return (
-    <Table
-      selectRows={true}
-      onChange={handleChange}
-      selectedValue={formData as RowData}
-    >
-      <TableHeader>
-        <TableRow>
-          <TableCell radiobutton={true}></TableCell>
-          {renderHeaders(tableHeaders)}
-        </TableRow>
-      </TableHeader>
-      <TableBody>
-        {calculatedDataList.map((datalist) => {
-          return (
-            <TableRow
-              key={JSON.stringify(datalist)}
-              rowData={rowAsValue(datalist)}
-            >
-              <TableCell radiobutton={true}>
-                <RadioButton
-                  name={datalist}
-                  onChange={() => {
-                    // Intentionally empty to prevent double-selection
-                  }}
-                  value={rowAsValueString(datalist)}
-                  checked={rowAsValueString(datalist) === JSON.stringify(formData) ? true : false}
-                  label={createLabelRadioButton(datalist)}
-                  hideLabel={true}
-                ></RadioButton>
-              </TableCell>
-              {renderRow(datalist)}
-            </TableRow>
-          );
-        })}
-      </TableBody>
-      {pagination && (
-        <TableFooter>
+    <FormControl component='fieldset'>
+      <FormLabel
+        component='legend'
+        classes={{ root: cn(classes.legend) }}
+        id={`${id}-label`}
+      >
+        <RenderLegend />
+      </FormLabel>
```

Comment:
Any specific reason you chose the mui one, and not [the one from our design system](https://designsystem.altinn.studio/?path=/story/components-fieldset--normal)? :thinking: If the one in the design system is not working right, we might need to look into that.

Reply:
Just because that was what was used for RadioGroup and CheckboxContainer. Tested with FieldSet from the design system now, but it doesnt seem to support screen reader? The screen reader works better with FormControl from mui, reading the label and "table", but with the FieldSet it doesnt read the label (but all the content of the table at the same time)

Reply:
Oh, huh. I'll create an issue for that in the design system then. Thanks for the research!